app/controllers/search_controller.rb

```diff
@@ -62,16 +71,20 @@ def tags
   end
 
   def chat_channels
-    search_user_id = if chat_channel_params[:user_id].present?
-                       [current_user.id, SiteConfig.mascot_user_id, chat_channel_params[:user_id]].reject(&:blank?)
-                     else
-                       [current_user.id]
-                     end
-    ccm_docs = Search::ChatChannelMembership.search_documents(
-      params: chat_channel_params.merge(user_id: search_user_id).to_h,
+    user_ids =
+      if chat_channel_params[:user_id].present?
+        [current_user.id, SiteConfig.mascot_user_id, chat_channel_params[:user_id]].reject(&:blank?)
+      else
+        [current_user.id]
+      end
+
+    result = Search::Postgres::ChatChannelMembership.search_documents(
```

Comment:
You'll notice I left everything in the `Search` namespace (the route, the controller, and the service naming). I originally started removing `chat_channels` from search all together since this behaves more like a _filter_ than a _search_.

However, I discovered there are some JS components that are using this endpoint as an "index". I started refactoring everything, but it quickly got out of hand and out of scope for what this PR is intended to address. I also realized it may actually be nice to leave everything in the search namespace so that when we re-do the search for Connect in the future, it's a smoother experience.

Reply:
> I started refactoring everything, but it quickly got out of hand and out of scope for what this PR is intended to address.

👍  good call taking the simpler approach in this PR 

With that said, since we have the `ChatChannelsController` and the `ChatChannelMembershipsController` each with index actions, I think the right move would be to move those JS components to them so we don't have multiple controllers performing the same actions. I also am always hesitant to leave code around with the idea _we might use it_. What do you think about making an Issue for it once this is merged so someone else can tackle the refactor at some point?

Reply:
https://github.com/forem/forem/issues/13277 🚀